src/App.tsx

```diff
@@ -1,26 +1,51 @@
+/* eslint-disable max-len */
+import { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+import { getTodos } from './api';
+
+import { useAppSelector } from './hooks/useAppSelector';
+import { todosSlice } from './features/todos';
```

Comment:
Ensure that `todosSlice` is correctly typed and imported. If you have a specific type for the actions, it might be beneficial to use it here to ensure type safety.